docs/project/release-notes.md

```diff
@@ -0,0 +1,219 @@
+maintenance improvements, a thorough redesign of the central APIs, and careful
+elimination of error leaks and memory leaks.
+
+There have been a large number of backwards incompatible changes due to these
+design improvements. Our hope is that we have fixed a significant portion of the
+issues and that future releases will have fewer breaking changes.
+
+## New Features
+
+### Asyncio Support
+
+We added full support for asyncio, including a new Python event loop that
+schedules tasks on the browser event loop, support for top level await in
+{any}`pyodide.runPythonAsync`, and implementations of `await` for {any}`JsProxy`
+and {any}`PyProxy`, so that it is possible to await a Python awaitable from
+Javascript and a Javascript thenable from Python. This allows seamless
+interoperability:
+```pyodide
+pyodide.runPython(`
+    async def test():
+        from js import fetch
+        # Fetch the Pyodide packages list
+        r = await fetch("packages.json")
+        data = await r.json()
+        # return all available packages
+        return data.dependencies.object_keys()
+`);
+let test = pyodide.globals.get("test");
+// test returns a coroutine, we can await the coroutine
+// from Javascript and it will schedule it on the Python event loop
+result = await test();
+console.log(result); // ["asciitree", "parso", "scikit-learn", ...]
+```
+(Added in PRs {pr}`880`, {pr}`1158`, {pr}`1170`)
+
+### Error Handling
+
+Errors can now be thrown in Python and caught in Javascript or thrown in
+Javascript and caught in Python.
+
+Support for this is integrated at the lowest level, so calls between Javascript
+and C functions behave as expected. The error conversion code is generated by C
+macros which makes implementing and debugging new logic dramatically simpler.
+
+```pyodide
+function jserror(){
+   throw new Error("ooops!");
+}
+pyodide.runPython(`
+   from js import jserror
+   from pyodide import JsException
+   try:
+       jserror()
+   except JsException as e:
+      print(str(args)) # prints "TypeError: ooops!"
+`);
+```
+(Added in PRs {pr}`1051` and {pr}`1080`)
+
+### Python "builtin" Modules implemented in Javascript
+
+It is now simple to add a Python module implemented in Javascript using
+{any}`pyodide.registerJsModule`:
+```pyodide
+let my_module = {
+    foo(x){
+        return x*x + 1;
+    },
+    bar(y, z){
+        return y*z + y + z;
+    }
+};
+pyodide.registerJsModule("my_mod", my_module);
+pyodide.runPython(`
+    from my_mod import foo, bar
+    foo(7) # 50
+    bar(9, 5) # 59
+`);
+```
+(Added in PR {pr}`1146`)
+
+### New Conversion APIs
+We added several new conversion APIs to give more explicit control over the
+foreign function interface. In particular, the goal was to make it easier for
+users to avoid leaking memory.
+
+For the basic use cases, we have {any}`PyProxy.toJs` and {any}`JsProxy.to_py`.
+We also added also "wrong-way" conversion functions {any}`pyodide.to_js` and
```

Comment:
It might be helpful to add slightly more explanation about these function as it might not be obvious for the reader.